app/javascript/components/KeyWordSearch.js

```diff
@@ -0,0 +1,63 @@
+import React, { Component } from 'react';
+import Button from 'react-bootstrap/lib/Button';
+import InputGroup from 'react-bootstrap/lib/InputGroup';
+import Form from 'react-bootstrap/lib/Form';
+import { faSearch } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+
+class KeyWordSearch extends React.Component{
+  constructor(props){
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this)
+    this.state = {
+      searchTerms: undefined,
+    };
+  };
+
+  handleSubmit(e){
+    // Prevent full page reload
+    e.preventDefault();
+
+    const searchTerm = e.target.elements.searchText.value.trim();
+    if(searchTerm){
+      // Need to check if search terms are empty and if prevstate is different
+      // From current state
+      this.setState(()=>{
+        return {
+          searchTerms:searchTerm
+        };
+
+      });
```

Comment:
my react must be too rusty, I don't understand what this second setState call is supposed to do -- it looks like you're undoing the previous setState call?

Reply:
SetState is asynchronous and React will batch multiple setState() calls into one for performance reasons. So to ensure state has actually change and avoid async issues, you can call it twice. 

Read more about it [here](https://reactjs.org/docs/state-and-lifecycle.html).